homeassistant/components/climate/ecobee.py

```diff
@@ -211,9 +211,9 @@ def turn_away_mode_off(self):
         """Turn away off."""
         self.data.ecobee.resume_program(self.thermostat_index)
 
-    def set_temperature(self, temperature):
+    def set_temperature(self, **data):
         """Set new target temperature."""
-        temperature = int(temperature)
+        temperature = int(data[ATTR_TEMPERATURE])
```

Comment:
@turbokongen 
1. the data attribute should already be coerced to an integer - no need to cast
2.
`temperature = data.get(ATTR_TEMPERATURE)`